go/chat/searcher.go

```diff
@@ -0,0 +1,90 @@
+package chat
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+)
+
+const pageSize = 300
+
+type Searcher struct {
+	globals.Contextified
+	utils.DebugLabeler
+}
+
+func NewSearcher(g *globals.Context) *Searcher {
+	labeler := utils.NewDebugLabeler(g.GetLog(), "searcher", false)
+	return &Searcher{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: labeler,
+	}
+}
+
+func (s *Searcher) Search(ctx context.Context, chatUI libkb.ChatUI, conversationID chat1.ConversationID, query string, maxHits int, maxMessages int) (res []chat1.RateLimit, err error) {
```

Comment:
I think it would be good to return the results from this function, and all make the chat UI optional (maybe by passing `nil`).

Reply:
That is, I should be able to invoke this without the UI and just get the results if I want.